packages/tc-schema-sdk/lib/updater.js

```diff
@@ -2,6 +2,23 @@ const EventEmitter = require('events');
 const fetch = require('node-fetch');
 const readYaml = require('./read-yaml');
 
+// e.g HasChild -> HAS_CHILD
+const toSnakeUpperCase = name =>
+	name
+		.replace(/([A-Z])/g, '_$1')
+		.replace(/^_|_$/g, '')
+		.toUpperCase();
+
+const readRelationships = schemaDirectory => {
+	if (!readYaml.isDirectory(schemaDirectory, 'relationships')) {
+		return [];
+	}
+	return readYaml.directory(schemaDirectory, 'relationships').map(rel => ({
+		...rel,
+		relationship: rel.relationship || toSnakeUpperCase(rel.name),
```

Comment:
Can you explain why this is needed? Is this supporting using the naming convention of the file to infer the relationship name, and therefore not expecting an explicit `relationship` property to exist? We don't support this for normal types, so I think we should either support it for both or for neither. 

I feel slightly in favour about always expecting the user to add a `relationship` property. It will be typed once, so not much effort to maintain, and makes things a bit less magic

Reply:
Refer to here https://trello.com/c/xZ6SbEGP/154-design-representations-of-custom-relationship-properties#comment-5dcd8ad9df28d53b62813a4f - `relationship` property in the root relationship type may not define, so I intended to attach it from type name e.g `HasChild` -> `HAS_CHILD`.

But I agree with we always expect the user adds  `relationship` property, and it should validate on tc-schema-validator